### ISSUE TITLE
Opening a smithy file included via a dependency fails

### ISSUE THREAD

**Issue by daddykotex** [disneystreaming/smithy-language-server#14]
given the following `smithy-build.json` file:

```json
{
    "mavenDependencies" : ["com.disneystreaming.smithy4s:smithy4s-protocol_2.13:latest.stable" ] 
}  
```

and a simple smithy definition file:

```smithy
namespace smithy4s.hello

use smithy4s.api#simpleRestJson

@simpleRestJson
service HelloWorldService {
  version: "1.0.0"
}
```

The VS code extension reports a problem coming from the smithy4s.smithy file (included in the downloaded jar). Note: this is possibly an error too. Clicking on the error will try to open a path that points to a jar.

The warning in question (may open another issue because I don't think it should be marked as an error):

![warning](https://user-images.githubusercontent.com/5230460/154142709-b2f06f3d-7e0a-46ea-b9f1-c7cf21159cb9.jpg)

Clicking on the issue attempt a jump to definition but can't open the file:

![error-after-click](https://user-images.githubusercontent.com/5230460/154142791-474b0a16-3e0c-4b8e-bf4f-f437c02f5fec.jpg)

A solution would be to extract the file from the jar, after the click, but before the file opening, and open the extracted file instead. I think scalameta/metals does something similar.

**Comment by daddykotex** [comment]
For example, here is a file, copied from a jar, into the .metals/readonly repository:

![example](https://user-images.githubusercontent.com/5230460/154152286-f7aa0fe1-67a3-4a1f-a782-d997ed58c9d1.jpg)

**Comment by daddykotex** [comment]
See: https://github.com/scalameta/metals/blob/58f06ff8a7322a0e2ed6bf167c70966b70ca49f9/metals/src/main/scala/scala/meta/internal/metals/debug/SourcePathAdapter.scala

**Comment by daddykotex** [comment]
Fixed in #16
